File: src/main/java/org/chousik/collection/builder/collectors/MusicGenreCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.MusicGenre;
import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

public class MusicGenreCollector extends EnumCollector<MusicGenre> {
    @Override
    public MusicGenre ask(String name, IValidator<String> validator) throws ScriptExecutionError {
        return askEnum(name, validator, MusicGenre::valueOf, MusicGenre::getValue);
    }
}

File: src/main/kotlin/collection/builder/collectors/MusicGenreCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.MusicGenre
import org.chousik.collection.validators.IValidator
import java.util.function.Function

class MusicGenreCollector : EnumCollector<MusicGenre>() {
    override fun ask(name: String, validator: IValidator<String?>): MusicGenre {
        return askEnum(
            name, validator,
            { value: String -> MusicGenre.valueOf(value) }, MusicGenre.value
        )
    }
}